### PR TITLE
Recipe cleanup

### PR DESCRIPTION
**Pull request by jakirkham**
* Drop old `x.x` syntax (has been deprecated for a while and is not needed)
* Switch `build` to `host` (this is how dependencies that are linked against should be listed)

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
LGTM, but we still need need a review from @rapidsai/ops-codeowners .

**Review by raydouglass — APPROVED** [review]

**Comment by pentschev** [comment]
Thanks @jakirkham , merging now.

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**
